mediacloud/mediawords/crawler/download/feed/ap.py

```diff
@@ -0,0 +1,207 @@
+        params.update(kwargs)
+        content_data = self._make_request(url,params)
+        return content_data
+
+    def _make_request(self,
+                      url: str,
+                      params: dict = None) -> str:
+        """Internal method for making API requests"""
+
+        retries = self.retry_limit
+
+        # Begin making request and retry up to retry limit
+        while retries:
+
+            logger.debug("Making request to {} with parameters {}".format(url,params))
+
+            try:
+                response = requests.get(url,params=params)
+            except Exception as e:
+                logger.warning("Encountered an exception while making request to {}. Exception info: {}".format(url,e))
+            else:
+                if response.status_code == 200:
+                    logger.debug("Successfully retrieved {}".format(url))
+                    return response.content
+                else:
+                    logger.warning("Received HTTP status code {} when fetching {}".format(response.status_code,url))
+
+            retries -= 1
+
+            if retries == 0:
+                logger.error("Could not fetch {} after {} attempts. Giving up.".format(url,self.retry_limit))
+                raise ValueError("Could not fetch {} after {} attempts. Giving up.".format(url,self.retry_limit))
+
+            wait_time = (self.retry_limit - retries) ** 2
+            logger.info("Exponentially backing off for {} seconds.".format(wait_time))
+            time.sleep(wait_time)
+
+
+def get_new_stories(db=None) -> list:
+    """This method fetches the latest items from the AP feed and returns a list of dicts.
+
+    Parameters:
+
+        db: If a db handle is passed in, this method will check for existing uuids in the
+        database and only fetch stories for uuids not present in the database. If no db handle
+        is passed, the script will return all stories (up to the max 100 without pagination)
+
+    Return Value:
+    Each returned dict includes the following keys:
+
+        guid: story id
+        url: public url to the story
+        publish_date
+        title: headline
+        description: headline_extended
+        text: story stripped of html tags
+        content: xml for item
+    """
+
+    api = AssociatedPressAPI()
+    feed_data = api.feed(page_size=100)
+    items = []  # list of dict items to return
+
+    for obj in feed_data['items']:
+
+        item = obj['item']
+        guid = item['altids']['itemid']
+        version = item['version']
+
+        # If DB handle was passed, check if this story has previously been retrieved (to avoid unnecessary API calls to content endpoint)
+        if db:
+            guid_exists = db.query("select 1 from stories s join media m using (media_id) where m.name = 'AP' and s.guid = %(a)s", {'a': guid}).hash()
+            if guid_exists:
+                logger.info('Story with guid: {} is already in the database -- skipping.')
```

Comment:
I would make this logger.debug().  my guess is that in production 98 of the 100 stories are going to be existing stories, so this will get very verbose.